### DISCUSSION TITLE
Filter localization exemple

### DISCUSSION THREAD

**Discussion by fbmfbm** [revolist/revogrid#651] [General]
I all,
I'm using revogrid in vueJs and i don't find localization information in the doc for the grid and filter options.
Any one know exemple for this ?
Thank's

**Comment by islarky88** [comment]
right now on typescript. there need to be changes with making some value for localization optional. so for our use, we needed to add the values even if it's default. 

```
get gridFilter(): ColumnFilterConfig {
    return {
      disableDynamicFiltering: true,
      localization: {
        // TODO: revogrid should make filterNames optional so we won't have to set this if we just need captions changed
        filterNames: {
          none: 'None',
          empty: 'Not set',
          notEmpty: 'Set',

          eq: 'Equal',
          notEq: 'Not equal',
          begins: 'Begins with',
          contains: 'Contains',
          notContains: 'Does not contain',

          eqN: '=',
          neqN: '!=',
          gt: '>',
          gte: '>=',
          lt: '<',
          lte: '<=',
        },
        // TODO: not all captions should be required. need changes in revogrid code
        captions: {
          save: 'Apply',
          reset: 'Clear',
          title: 'Filter by condition',
          cancel: 'Cancel',
        },
      },
      multiFilterItems: {},
      include: ['contains', 'notContains', 'eq', 'notEq', 'begins'],
    };
  }
```

**Comment by ghost** [comment]
The filters do not work. If I select a value and a filter conditions and then select Apply, nothing happens. Also no error on page. I am using Vue 2 and Javascript

const filterConfig = {
    disableDynamicFiltering: true,
    localization: {
        // REVO TODO: revogrid should make filterNames optional so we won't have to set this if we just need captions changed
        filterNames: {
            none: 'None',
            empty: 'Not set',
            notEmpty: 'Set',

            eq: 'Equal',
            notEq: 'Not equal',
            begins: 'Begins with',
            contains: 'Contains',
            notContains: 'Does not contain',

            eqN: '=',
            neqN: '!=',
            gt: '>',
            gte: '>=',
            lt: '<',
            lte: '<=',
        },
        // REVO TODO: not all captions should be required. need changes in revogrid code
        captions: {
            save: 'Apply',
            reset: 'Clear',
            title: 'Filter by condition',
            cancel: 'Cancel',
        },
    },
    multiFilterItems: {},
    include: ['contains', 'notContains', 'eq', 'notEq', 'begins']
};

grid.filter = filterConfig;
